src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java

```diff
@@ -185,4 +199,109 @@
         r.assertBuildStatusSuccess(r.waitForCompletion(b4));
     }
 
+    @Issue("JENKINS-34005")
+    @Test public void triggersProperty() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        // Verify the base case behavior.
+        p.setDefinition(new CpsFlowDefinition("echo 'foo'"));
+
+        assertTrue(p.getTriggers().isEmpty());
+
+        r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+
+        // Make sure the triggers are still empty.
+        assertTrue(p.getTriggers().isEmpty());
+
+        // Now add a trigger.
+        p.setDefinition(new CpsFlowDefinition("properties([[$class: 'PipelineTriggersJobProperty',\n"
```

Comment:
...once we've gotten [JENKINS-29922](https://issues.jenkins-ci.org/browse/JENKINS-29922) fully landed, yeah - I think we're still going to want the `properties` step containing `pipelineTriggers(...)` and friends, though, since we wipe out any pre-existing properties currently...